controller/admin_controller.php

```diff
@@ -0,0 +1,296 @@
+	{
+		$this->config = $config;
+		$this->db = $db;
+		$this->language = $language;
+		$this->log = $log;
+		$this->request = $request;
+		$this->template = $template;
+		$this->user = $user;
+		$this->phpbb_root_path = $root_path;
+		$this->php_ext = $php_ext;
+	}
+
+	/**
+	 * Display the options a user can configure for this extension
+	 *
+	 * @return null
+	 * @access public
+	 */
+	public function display_options()
+	{
+		// Create a form key for preventing CSRF attacks
+		add_form_key('acp_phpbb_ideas_settings');
+
+		$this->new_config = $this->config;
+		$this->cfg_array = ($this->request->is_set('config')) ? $this->request->variable('config', array('' => ''), true) : $this->new_config;
+
+		// Check if selected user exists
+		$this->check_ideas_topics_poster();
+
+		// Configuration options to list through
+		$display_vars = array(
+			'ideas_forum_id',
+			'ideas_poster_id',
+			'ideas_base_url',
+			'ideas_forum_setup',
+		);
+		$this->set_config_options($display_vars);
+
+		// Output relevant page
+		$this->template->assign_vars(array(
+			'IDEAS_POSTER'		=> $this->get_ideas_topics_poster(),
+			'IDEAS_BASE_URL'	=> ($this->config['ideas_base_url']) ?: '',
+
+			'S_FORUM_SELECT_BOX'	=> $this->select_ideas_forum(),
+			'S_IDEAS_FORUM_ID'		=> !empty($this->config['ideas_forum_id']),
+
+			'U_ACTION'			=> $this->u_action,
+			'U_FIND_USERNAME'	=> append_sid("{$this->phpbb_root_path}memberlist.{$this->php_ext}", 'mode=searchuser&amp;form=acp_phpbb_ideas_settings&amp;field=ideas_poster_id&amp;select_single=true'),
+		));
+	}
+
+	/**
+	 * Check if Ideas bot user exists
+	 *
+	 * @return null
+	 * @access public
+	 */
+	public function check_ideas_topics_poster()
+	{
+		if ($this->request->is_set_post('submit'))
+		{
+			if (!check_form_key('acp_phpbb_ideas_settings'))
+			{
+				trigger_error($this->language->lang('FORM_INVALID') . adm_back_link($this->u_action));
+			}
+
+			$sql = 'SELECT user_id
+				FROM ' . USERS_TABLE . "
+				WHERE username_clean = '" . $this->db->sql_escape(utf8_clean_string($this->cfg_array['ideas_poster_id'])) . "'";
+			$result = $this->db->sql_query($sql);
+			$user_id = (int) $this->db->sql_fetchfield('user_id');
+			$this->db->sql_freeresult($result);
+
+			if (!$user_id)
+			{
+				trigger_error($this->language->lang('NO_USER') . adm_back_link($this->u_action));
+			}
+			else
+			{
+				// If selected user does exist, reassign the config value to its ID
+				$this->cfg_array['ideas_poster_id'] = $user_id;
+			}
+		}
+	}
+
+	/**
+	 * Get ideas poster bot username
+	 *
+	 * @return string Ideas bot username
+	 * @access public
+	 */
+	public function get_ideas_topics_poster()
+	{
+		$ideas_poster_id = (int) $this->config['ideas_poster_id'];
+		$sql = 'SELECT username FROM ' . USERS_TABLE . '
+			WHERE user_id = ' . $ideas_poster_id;
+		$this->db->sql_query($sql);
+		$username = $this->db->sql_fetchfield('username');
+		$username = ($username !== false) ? $username : '';
```

Comment:
Instead of reassigning, you could just return this:
`return ($username !== false) ? $username : '';`